provisioner/aws-pricing.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+var Promise = require('promise');
+var lodash = require('lodash');
+var util = require('util');
+var Cache = require('../cache');
+var assert = require('assert');
+
+
+function fetchPricing(ec2) {
+  var that = this;
+  var startDate = new Date();
+  startDate.setHours(startDate.getHours() - 2);
+  var requestObj = {
+    StartTime: startDate,
+    Filters: [{
+      Name: 'product-description',
+      Values: ['Linux/UNIX'],
+    }],
+  }
+  var p = ec2.describeSpotPriceHistory(requestObj);
+
+  p = p.then(function(pricing) {
+    var regions = Object.keys(pricing);
+    var fixed = {};
+
+    // Get rid of the key we don't care about
+    regions.forEach(function(region) {
+      fixed[region] = pricing[region].SpotPriceHistory;
+    });
+
+    return new AwsPricing(ec2.regions, fixed);
+  })
+
+  return p;
+}
+
+module.exports = fetchPricing;
+
+function AwsPricing(regions, pricing) {
+  this.__pricing = pricing;
+  this.regions = regions;
+}
+
+
+/**
+ * Get the pricing dictionary... This is a little ugly
+ * but it works... Ideas welcomed
+ */
+AwsPricing.prototype.get = function () {
```

Comment:
`priceHistoryByRegion` if it's naming you're looking for...

You should probably return a clone. If you maintain state... I doubt performance is an issue here, granted if you have a tonn of calls to it, you might want to do something else. (Nothing, or try with `object.freeze`)